esphome/components/fan/__init__.py

```diff
@@ -217,10 +218,21 @@ async def fan_turn_on_to_code(config, action_id, template_arg, args):
     return var
 
 
-@automation.register_action("fan.cycle_speed", CycleSpeedAction, FAN_ACTION_SCHEMA)
+@automation.register_action(
+    "fan.cycle_speed",
+    CycleSpeedAction,
+    FAN_ACTION_SCHEMA.extend(
+        {
+            cv.Optional(CONF_OFF_SPEED_CYCLE, default=True): cv.boolean,
+        }
+    ),
```

Comment:
In that case, we will just reimplement, not extend
```suggestion
    maybe_simple_id(
        {
            cv.Required(CONF_ID): cv.use_id(Fan),
            cv.Optional(CONF_OFF_SPEED_CYCLE, default=True): cv.boolean,
        }
    ),
```

Reply:
Thanks.  I'm primarily a web developer, and don't know all the features of C++.  Sometimes I don't see the obvious solution, and this project is one of the most complex things I've ever seen.  I appreciate your consideration.